newIDE/app/src/SceneEditor/index.js

```diff
@@ -1028,6 +1029,43 @@ export default class SceneEditor extends React.Component<Props, State> {
     };
     return (
       <div style={styles.container}>
+        <UseCompoundCommandHook
+          commandName={'EDIT_OBJECT'}
+          command={{
+            displayText: t`Edit object...`,
+            enabled: true,
+            generateOptions: () =>
+              enumerateObjects(project, layout).containerObjectsList.map(
+                item => ({
+                  text: item.object.getName(),
+                  value: item.object,
+                  iconSrc: ObjectsRenderingService.getThumbnail.bind(
+                    ObjectsRenderingService
+                  )(this.props.project, item.object),
+                  handler: () =>
+                    (this.props.onEditObject || this.editObject)(item.object),
+                })
+              ),
+          }}
+        />
+        <UseCompoundCommandHook
+          commandName={'EDIT_OBJECT_VARIABLES'}
+          command={{
+            displayText: t`Edit object variables...`,
+            enabled: true,
+            generateOptions: () =>
+              enumerateObjects(project, layout).containerObjectsList.map(
```

Comment:
if we move this to a hook/component "UseSceneEditorCommands", we can probably factor this function in something called `generateLayoutObjectsOption({project, layout, onChoose: (object) => this.editObjectVariables(object)})`.

(just to avoid this tiny bit of repetition, though this would be really worth factoring if there are more than 2 usages of it).

Reply:
Done. I've moved the commands to a separate file like in Mainframe, seems cleaner.